website_local_fonts/static/src/js/snippets.options.js

```diff
@@ -0,0 +1,270 @@
+/* Copyright 2023 Onestein - Anjeel Haria
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). */
+odoo.define("website_local_fonts.editor.snippets.options", function (require) {
+    "use strict";
+
+    const website_editor_snippet_options = require("website.editor.snippets.options");
+    const FontFamilyPickerUserValueWidget =
+        website_editor_snippet_options.FontFamilyPickerUserValueWidget;
+    var core = require("web.core");
+    var Dialog = require("web.Dialog");
+    const weUtils = require("web_editor.utils");
+    var _t = core._t;
+    var qweb = core.qweb;
+    var options = require("web_editor.snippets.options");
+
+    website_editor_snippet_options.FontFamilyPickerUserValueWidget.include({
+        xmlDependencies: (
+            FontFamilyPickerUserValueWidget.prototype.xmlDependencies || []
+        ).concat(["/website_local_fonts/static/src/xml/website.editor.xml"]),
+        events: _.extend({}, FontFamilyPickerUserValueWidget.prototype.events || {}, {
+            "click .o_we_add_local_font_btn": "_onAddLocalFontClick",
+            "click .o_we_delete_local_font_btn": "_onDeleteLocalFontClick",
+        }),
+        /**
+         * @override
+         */
+        start: async function () {
+            await this._super(...arguments);
+            $(this.menuEl).empty();
```

Comment:
Do you need to empty the element here? Is possible to append the local fonts instead?

Reply:
No, as the delete button shown next to the fonts is done based on the index of the font elements and this index cannot be fetched for elements already added